### PR TITLE
SHADHO Packaging Test

### PR DESCRIPTION
**Pull request by tphung3**
This packaging test is not yet finished. Somehow the work queue worker still survives and prints out gibberish after receiving the SIGINT signal sent from the shell that spawns it.

### CODE REVIEW

**Comment by dthain** [comment]
A couple of thoughts -
- If you can figure out how to run the worker from the top-level script instead of indirectly via SHADHO, then it will be easier to control what's going on.
- Terminating the worker is a good idea, however, it looks like the other integration tests don't bother doing so.  Perhaps it isn't necessary?
- Consider using the --single-shot mode of the worker, which causes it to shutdown after the first disconnection.
-

**Comment by dthain** [comment]
- Or just use SIGKILL on the worker.
-

**Comment by tphung3** [comment]
Worker is now created directly and uses --single-shot to terminate after first disconnection.
The shadho packaging test should be okay to merge now.

File: shadho-test.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+conda create --name shadho-test -y
+conda activate shadho-test
+
+#install ndcctools and shadho
+conda install -c conda-forge ndcctools -y
+conda install pip -y	#shadho is installed through pip
+python -m pip install shadho
+
+#run work_queue_worker. This command is heuristic and based on shadho code.
+$CONDA_PREFIX/bin/work_queue_worker -M sin_ex-tphung --cores 1 --single-shot &
```
Comment (dthain) [inline shadho-test.sh:20]: This is going to pop up in the catalog daily.  Let's call it shadho-wq-packaging-test or something like that.

File: shadho-test.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#kill the worker, to be sure
+kill -2 $SHADHO_WORKER_PID
+
+#deactivate and remove the created environment
+conda deactivate
+conda remove --name shadho-test --all -y
+
+#kill the shell that runs this script
+kill -2 $$
```
Comment (dthain) [inline shadho-test.sh:40]: This will result in a strange exit status, it's better to just fall off the end.

File: shadho-test.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export SHADHO_WORKER_PID=$!
+
+#run manager with simple application
+python wq_sin.py
+
+#remove resulting files from shadho application
+rm results.json results.json.bak.1 
+rm shadho_master.debug shadho_master.log
+
+#kill the worker, to be sure
```
Comment (dthain) [inline shadho-test.sh:32]: How about kill -9?

**Review by dthain — CHANGES_REQUESTED** [review]

File: shadho-test.sh (hunk 3 above)
Comment (tphung3) [reply shadho-test.sh:32]: kill -9 may not allow worker to clean its workspace I think. Is there any reason behind sending -9 signal?

Comment (dthain) [reply shadho-test.sh:32]: The whole test gets run in some kind of VM/container anyway, so I'm not concerned about cleanup at this level...

File: shadho-test.sh (hunk 1 above)
Comment (tphung3) [reply shadho-test.sh:20]: Shadho automatically appends user name to the project name, so shadho-wq-packaging-test is renamed to shadho-wq-packaging-test-tphung, for example (A bit weird to let user name exposed).

File: shadho-test.sh [hunk 4]
```diff
@@ -8,16 +8,16 @@ CONDA_BASE=$(conda info --base)
...
+conda activate shadho-wq-packaging-test
 
 #install ndcctools and shadho
 conda install -c conda-forge ndcctools -y
 conda install pip -y	#shadho is installed through pip
 python -m pip install shadho
 
 #run work_queue_worker. This command is heuristic and based on shadho code.
-$CONDA_PREFIX/bin/work_queue_worker -M sin_ex-tphung --cores 1 --single-shot &
+$CONDA_PREFIX/bin/work_queue_worker -M shadho-wq-packaging-test-tphung --cores 1 --single-shot &
```
Comment (dthain) [inline shadho-test.sh:20]: Or maybe just shadho-wq-packaging-test-${USER} ?

**Pull request merged**